yaksh/views.py

```diff
@@ -390,26 +390,27 @@ def prof_manage(request, msg=None):
         return my_redirect('/exam/login')
     if not is_moderator(user):
         return my_redirect('/exam/')
-    courses = Course.objects.filter(creator=user, is_trial=False)
-
+    courses = Course.objects.filter(Q(creator=user) | Q(teachers=user),
+                                    is_trial=False)
     trial_paper = AnswerPaper.objects.filter(
         user=user, question_paper__quiz__is_trial=True,
         course__is_trial=True
     )
     if request.method == "POST":
         delete_paper = request.POST.getlist('delete_paper')
         for answerpaper_id in delete_paper:
-            answerpaper = AnswerPaper.objects.get(id=answerpaper_id)
-            qpaper = answerpaper.question_paper
-            answerpaper.course.remove_trial_modules()
-            answerpaper.course.delete()
-            if qpaper.quiz.is_trial:
-                qpaper.quiz.delete()
-            else:
-                if qpaper.answerpaper_set.count() == 1:
+            answerpaper = AnswerPaper.objects.filter(id=answerpaper_id)
```

Comment:
Searching from id will always give only one object, why filter here?

Reply:
I think he wants to avoid ```try...except``` in case the object does not exists.